### DISCUSSION TITLE
Make dragging and dropping a new background only affect the current difficulty you're editing.

### DISCUSSION THREAD

**Discussion by AFasterSlowpoke** [ppy/osu#30435] [Ideas] [answered]
When I drag an image to the osu lazer editor to set the background, it changes the background for all difficulties, instead of just the difficulty i'm currently editing.

I think that it would improve the osu lazer editor if dragging and dropping a new background only changed the current difficulty's background, with a button to set the current difficulty's background to all other difficulties.

**Comment by peppy** [comment]
@AFasterSlowpoke Just to confirm, it doesn't seem to update the background for other difficulties, but it does **remove** it from the other difficulties. Is this what you're seeing?

**Reply by AFasterSlowpoke** [reply]
Sorry for the late response, but I think that when you add a new bg by drag+dropping, all other backgrounds in the mapset get deleted.(hopefully this is clear enough)

Scenario 1:
- upload png background to a diff in a maspet
- its called "bg.png"
- upload another png background to the diff
- original bgs in map folder gets deleted, the new background is called "bg.png", all diffs in mapset have the same bg if they previously had png background (this also happens if both bgs are jpgs or otherwise the same file type)

Scenario 2:
- upload png background to a diff in a maspet
- its called "bg.png"
- upload jpg background to a diff
- original bgs in map folder gets deleted, the new background is called "bg.jpg", other diffs in mapset appear to not have a bg if their og backgrounds had a different file type than jpg.

Expected behaviour:
- upload example1.png to Easy diff
- only easy diff has its bg updated, rest of mapset retains their original backgrounds
- upload example2.jpg to Hard diff
- only hard diff has its bg updated, rest of mapset retains their original backgrounds
- you get the drill

**Reply by AFasterSlowpoke** [reply]
**my idea** is to make it so it is like stable, it doesn't delete the backgrounds upon bg change, and it calls them by their og file (e.g. if you drag and drop "example.png", the bg will be named "example.png" instead of "bg.png".) it should also warn you about unused backgrounds and files when you upload the map (when you can do that in lazer) and give you the option to delete them like in stable.
i now use lazer as my default mapping+playing client as it has a lot of new useful things, but there are some things i have to still use stable for because lazer editor does lack some features. one day, i hope i can completely abandon the stable client

**Reply by peppy** [reply, marked as answer]
Thanks for confirmation, tracking here https://github.com/ppy/osu/issues/30806
